lib/recording/RecordingEntrySession.js

```diff
@@ -369,16 +427,72 @@ class RecordingEntrySession {
 
     }
 
+ /*   verifyStreamMatchPreviousFlavorSet(newEntryObj) {
+        return qio.exists(this.recordingSessionPath)
+            .then((sessionExist)=> {
+                if (!sessionExist) {
+                    return Q.resolve(true);
+                }
+                else {
+                    let newFlavorNames = _.map(newEntryObj.flavorsObjArray, function(newFlavorDir) {
+                        return newFlavorDir.name;
+                    });
+                    let pattern = /^\d+$/;
+                    return fsUtils.isSameFlavorSet(this.recordingSessionPath, newFlavorNames, [newEntryObj.recordedEntryId], pattern);
+                }
+            })
+            .catch((err) => {
+                this.logger.error(`failed to verify that current flavors match previous recording. Error: ${err}`);
+                // since failed to verify match choosing fallback which is assuming no change in flavor set
+                return Q.resolve(true);
+            })
+    }*/
+
+    verifyFlavorSetMatchPlaylistFlavors(newEntryObj) {
+        let playlistFilePath = path.join(this.recordingSessionPath, 'playlist.json');
+        return qio.read(playlistFilePath)
+            .catch((err) =>{
+                if (err instanceof Error && err.code !== 'ENOENT') {
+                    this.logger.error(`Failed to read [${this.recordingSessionPath}]. Error: ${ErrorUtils.error2string(err)}`);
+                    return Q.reject(err);
+                }
+                else
+                    return Q.resolve();
+            })
+            .then((data) => {
+                let playlistJSON = JSON.parse(data);
+                return _.map(playlistJSON.sequences, (sequence) => {
+                    return sequence.id;
+                });
+            })
+            .then((flavorsFromPlaylist)=> {
+                let newFlavorNames = _.map(newEntryObj.flavorsObjArray, function(newFlavorDir) {
+                    return newFlavorDir.name;
+                });
+                // check if flavors match current provisioned flavors
+                if (flavorsFromPlaylist.length != newFlavorNames.length) {
```

Comment:
not exactly... the difference returns the set of items from array (other) that are not on the right so _.difference( [32], [32,33]) will return an empty array, thus wrong to decide that flavors didn't change. Question is whether this is possible without changing transcoding profile: [32,33] => [32,34] if this is impossible then it is enough to compare arrays size. 
_.difference(array, *others) 
returns the values from array that are not present in the other arrays.